pkg/apis/pipeline/v1beta1/merge_test.go

```diff
@@ -117,3 +117,379 @@ func TestMergeStepsWithStepTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeStepOverrides(t *testing.T) {
+	tcs := []struct {
+		name          string
+		steps         []Step
+		stepOverrides []TaskRunStepOverride
+		want          []Step
+		wantErr       bool
```

Comment:
nit: the `wantErr` field does not seem to be used, so remove it. (For error cases, a lot of times its a bit more readable to have a separate Error tests function e.g. TestMergeStepOverridesError)

Reply:
nice catch thanks! done